redskyctl/internal/commander/commander.go

```diff
@@ -64,6 +65,51 @@ func (s *IOStreams) OpenFile(filename string) (io.ReadCloser, error) {
 	return os.Open(filename)
 }
 
+// YAMLReader returns a resource node reader for the named file.
+func (s *IOStreams) YAMLReader(filename string) kio.Reader {
+	if filename == "-" {
+		path := "/dev/stdin"
+		if namedReader, ok := s.In.(interface{ Name() string }); ok {
+			path = namedReader.Name()
+		}
+		return &kio.ByteReader{
+			Reader:         s.In,
+			SetAnnotations: map[string]string{kioutil.PathAnnotation: path},
+		}
+	}
+
+	return &kio.ByteReader{
+		Reader:         &fileReader{Filename: filename},
+		SetAnnotations: map[string]string{kioutil.PathAnnotation: filename},
+	}
```

Comment:
thoughts on reducing the returns?

```suggestion
	var reader io.Reader = &fileReader{Filename: filename}
	annotations := map[string]string{kioutil.PathAnnotation: filename}

	if filename == "-" {
  		path := "/dev/stdin"
  		if namedReader, ok := s.In.(interface{ Name() string }); ok {
  			path = namedReader.Name()
  		}
  		reader = s.In
  		annotations[kioutil.PathAnnotation] = path
  	}
 
  	return &kio.ByteReader{
  		Reader:         reader,
  		SetAnnotations: annotations,
  	}
```

Reply:
Actually, in thinking about it, we don't want the path set to `/dev/stdin` because then it will try to resolve relative file paths against `/dev/`.